pygerduty/__init__.py

```diff
@@ -391,11 +395,20 @@ def __init__(self, *args, **kwargs):
         self.log_entries = LogEntries(self.pagerduty, self)
         self.notes = Notes(self.pagerduty, self)
 
-    def _do_action(self, verb, requester_id, **kwargs):
+    def _do_action(self, verb, requester_id, method='PUT', **kwargs):
         path = '{0}/{1}/{2}'.format(self.collection.name, self.id, verb)
         data = {'requester_id': requester_id}
         data.update(kwargs)
-        self.pagerduty.request("PUT", path, data=json.dumps(data))
+        return self.pagerduty.request(method, path, data=json.dumps(data))
+
+    def has_subject(self):
+        has_subject = False
+        try:
+            self.trigger_summary_data.subject  # try to access the subject
+            has_subject = True
+        except:
```

Comment:
In general I like to prevent naked exceptions. I imagine you're looking to catch an `AttributeError` when the `subject` doesn't exist on `self.trigger_summary_data`. Could this be simplified to 

```
def has_subject(self):
    return hasattr(this.trigger_summary_data, "subject")
```